mts/platform/deviceinfo/deviceinfoprovider.cpp

```diff
@@ -44,27 +45,21 @@ using namespace meegomtp1dot0;
 DeviceInfoProvider::DeviceInfoProvider()
     : m_batteryStatus(new BatteryStatus(this))
 {
-    QDeviceInfo di;
-
+    DeviceInfo deviceInfo;
     SsuDeviceInfo ssuDeviceInfo;
 
     m_serialNo = ssuDeviceInfo.deviceUid();
-
-    m_deviceVersion = QString("%1 HW: %2").arg(di.version(QDeviceInfo::Os))
-                      .arg(di.version(QDeviceInfo::Firmware));
-
-    m_manufacturer = di.manufacturer().isEmpty() ? m_manufacturer : di.manufacturer();
-    m_model = di.model().isEmpty() ? m_model : di.model();
+    m_deviceVersion = QString("%1 HW: %2").arg(deviceInfo.osVersion()).arg(deviceInfo.firmwareVersion());
```

Comment:
Should this be actually os version here and in api definitions of the parent class? MTP is used for mp3 players, cameras etc and there the "firmware" is sort of the version of the software it's running. This gets used for MTPResponder but it doesn't have other os version info used. Now, why would the remote end be interested in hardware adaptation instead of the version of the OS it's running.

Reply:
There is a chance that it is used by initiator/pc for a) selecting workarounds for issues the responder/phone presumably has, and b) it could be shown to user too. For purposes of (a) one might argue that identifier buteo-mtp version might be more appropriate, but less so for (b). Additionally (a) would imply that there can be initiators out there that cease to work if we make radical changes -> sort of "it is what it is" situation again.

Reply:
How about we just remove the "HW: ..." part out of the string? Rationale: would be simpler, pretty confident there's no other end that would have any workarounds for buteo-mtp, and the hw adaptation version shouldn't matter for the workarounds if something some day would think of adding such.

Reply:
> pretty confident there's no other end that would have any workarounds for buteo-mtp

Personally I'm assuming that all mtp initiators and responders are trying to work around peculiarities of the other end. 

But whether it is done based on this version string is another question. Probably not. Rolling the dice as suggested.

Reply:
Workarounds sure, but less likely for sailfish and especially by using the hardware adaptation version :)